App/Containers/ThreadInvite.js

```diff
@@ -0,0 +1,154 @@
+        <View>
+          <Text style={style.status}>{title}</Text>
+          <ActivityIndicator size='large' color='#000000' animating={this.state.status !== 'success'} />
+          <View style={style.buttonMargin} />
+          <Button
+            style={style.button}
+            title='Continue'
+            accessibilityLabel='continue'
+            onPress={this.continue.bind(this)}
+          />
+          <Button
+            style={style.button}
+            title='Cancel'
+            accessibilityLabel='cancel'
+            onPress={this.cancel.bind(this)}
+          />
+        </View>
+      </View>
+    )
+  }
+
+  renderSuccess () {
+    // TODO: redirect the user to the newly joined thread
+    return (
+      <View style={[photosStyle.container, style.container]}>
+        <View>
+          <Text style={style.status}>Success!</Text>
+          <View style={style.buttonMargin} />
+          <Button
+            style={style.button}
+            title='Continue'
+            accessibilityLabel='continue'
+            onPress={this.continue.bind(this)}
+          />
+        </View>
+      </View>
+    )
+  }
+
+  renderError (message) {
+    return (
+      <View style={[photosStyle.container, style.container]}>
+        <View>
+          <Text style={style.status}>ERROR</Text>
+          <Text style={style.message}>
+            {message}
+          </Text>
+          <Button
+            style={style.button}
+            title='Continue'
+            accessibilityLabel='continue'
+            onPress={this.continue.bind(this)}
+          />
+        </View>
+      </View>
+    )
+  }
+
+  render () {
+    if (!this.state.key) {
+      return this.renderError('There was an issue pairing with your new device. This may be caused by network connectivity or other issues. Please try again. If it continues, please report the issue with Textile.')
+    } else if (this.props.threads.some((t) => t.id === this.state.key)) {
+      // the thread already exists
+      return this.renderError('You are already a member of the thread you are trying to join.')
+    } else if (this.state.status === 'confirmed') {
+      // TODO: should render that we are waiting to start or connect to the network
+      return this.renderPairing('JOINING')
+    } else if (this.state.status === 'added') {
+      return this.renderSuccess()
+    }
+    return this.renderConfirm()
+  }
+}
+
+const mapStateToProps = state => {
+  const online = state.ipfs && state.ipfs.online && state.ipfs.online ? state.ipfs.online : false
+  const nodeState = state.ipfs && state.ipfs.nodeState ? state.ipfs.nodeState.state === 'started' : false
+  return {
+    threads: state.threads && state.threads.threadItems ? state.threads.threadItems : [],
+    online: nodeState && online
+  }
+}
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    addThreadRequest: (name, id) => { dispatch(ThreadsAction.addThreadRequest(name, id)) },
```

Comment:
Just looking at textile-go work and realizing this is calling the wrong method. We want to call this method here, https://github.com/textileio/textile-go/pull/158/files#diff-95a702539b8bc16ea2eb14e093c5555aR290

Reply:
fixed now